live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/domain/event/IngestNewNodeEventHandlerTest.java

```diff
@@ -26,91 +26,43 @@
-    private static final boolean NODE_IS_FILE = false;
-    private static final long NODE_CREATED_AT = 1_690_000_000_050L;
-    private static final NodeProperty<String> NODE_TITLE = new NodeProperty<>("cm:title", "some title");
-    private static final Set<NodeProperty<?>> NODE_PROPERTIES = Set.of(NODE_TITLE);
-    @Captor
-    ArgumentCaptor<UpdateNodeMetadataEvent> updateNodeMetadataEventCaptor;
-    private final EventPublisher eventPublisher = mock();
-    private final IngestNewNodeEventHandler ingestNewNodeEventHandler = new IngestNewNodeEventHandler(eventPublisher);
+    @Mock
+    IngestNewNodeEvent ingestNewNodeEvent;
+    @Mock
+    UpdateNodeMetadataEvent updateNodeMetadataEvent;
+    @Mock
+    UpdateNodeEventMapper updateNodeEventMapper;
+    @Mock
+    EventPublisher eventPublisher;
+    @InjectMocks
+    IngestNewNodeEventHandler ingestNewNodeEventHandler;
 
     @Test
-    void shouldSetNewlyCreatedNodeMetadataProperties()
+    void nodeWithoutContent_PublishMetadata()
     {
         // given
-        Node node = new Node(
-                NODE_ID,
-                NODE_NAME,
-                NODE_PRIMARY_ASSOC_Q_NAME,
-                NODE_TYPE,
-                NODE_CREATED_BY_USER_WITH_ID,
-                NODE_MODIFIED_BY_USER_WITH_ID,
-                NODE_ASPECT_NAMES,
-                NODE_IS_FILE,
-                NODE_IS_FOLDER,
-                NODE_CREATED_AT,
-                NODE_PROPERTIES);
-
-        IngestNewNodeEvent event = new IngestNewNodeEvent(
-                EVENT_TIMESTAMP,
-                node);
-
-        Set<NodeProperty<?>> expectedNodePropertiesToSet = Set.of(
-                NAME.withValue(NODE_NAME),
-                PRIMARY_ASSOC_Q_NAME.withValue(NODE_PRIMARY_ASSOC_Q_NAME),
-                TYPE.withValue(NODE_TYPE),
-                CREATED_BY_USER_WITH_ID.withValue(NODE_CREATED_BY_USER_WITH_ID),
-                MODIFIED_BY_USER_WITH_ID.withValue(NODE_MODIFIED_BY_USER_WITH_ID),
-                ASPECTS_NAMES.withValue(NODE_ASPECT_NAMES),
-                IS_FILE.withValue(NODE_IS_FILE),
-                IS_FOLDER.withValue(NODE_IS_FOLDER),
-                CREATED_AT.withValue(NODE_CREATED_AT),
-                NODE_TITLE);
+        given(updateNodeEventMapper.map(ingestNewNodeEvent)).willReturn(updateNodeMetadataEvent);
 
         // when
-        ingestNewNodeEventHandler.handle(event);
+        ingestNewNodeEventHandler.handle(ingestNewNodeEvent);
 
         // then
-        verify(eventPublisher).publishMessage(updateNodeMetadataEventCaptor.capture());
-        UpdateNodeMetadataEvent updateNodeMetadataEvent = updateNodeMetadataEventCaptor.getValue();
-
-        assertContainsSameElements(expectedNodePropertiesToSet, updateNodeMetadataEvent.getMetadataPropertiesToSet().values());
-        assertTrue(updateNodeMetadataEvent.getMetadataPropertiesToUnset().isEmpty(), "There should be no properties to unset");
+        verify(eventPublisher).publishMessage(updateNodeMetadataEvent);
```

Comment:
Here also IMO we should choose between Mockito's `when/verify` or `given/then`

Reply:
This is fixed in the next PR: #19